ci/dev/vscode.patch

```diff
@@ -3199,6 +3275,84 @@ index 94e7e7a4bac154c45078a1b5034e50634a7a43af..8164200dcef1efbc65b50eef9c270af3
  				this._filenameKey.set(value ? basename(value) : null);
  				this._dirnameKey.set(value ? dirname(value).fsPath : null);
  				this._pathKey.set(value ? value.fsPath : null);
+diff --git a/src/vs/workbench/contrib/remote/browser/remote.ts b/src/vs/workbench/contrib/remote/browser/remote.ts
+index 98573a206f14928fc3fdf18fe927cb75034e4ad1..1430666aa94f941bda086df503fec8b35aa2b25f 100644
+--- a/src/vs/workbench/contrib/remote/browser/remote.ts
++++ b/src/vs/workbench/contrib/remote/browser/remote.ts
+@@ -730,6 +730,7 @@ class RemoteAgentConnectionStatusListener implements IWorkbenchContribution {
+ 		@IContextKeyService contextKeyService: IContextKeyService
+ 	) {
+ 		const connection = remoteAgentService.getConnection();
++		const SHOW_POPUP_ON_ATTEMPT = 2 // aka third attempt
+ 		if (connection) {
+ 			let visibleProgress: VisibleProgress | null = null;
+ 			let lastLocation: ProgressLocation.Dialog | ProgressLocation.Notification | null = null;
+@@ -793,33 +794,47 @@ class RemoteAgentConnectionStatusListener implements IWorkbenchContribution {
+ 					disposableListener.dispose();
+ 					disposableListener = null;
+ 				}
++				let suppressPopup = (typeof e.connectionAttempt == 'number' && e.connectionAttempt < SHOW_POPUP_ON_ATTEMPT)
++				let forceDialog = (typeof e.connectionAttempt == 'number' && e.connectionAttempt == SHOW_POPUP_ON_ATTEMPT)
+ 				switch (e.type) {
+ 					case PersistentConnectionEventType.ConnectionLost:
+-						if (!visibleProgress) {
+-							visibleProgress = showProgress(ProgressLocation.Dialog, [reconnectButton, reloadButton]);
++						if (suppressPopup) {
++							hideProgress()
```

Comment:
Why are these calls necessary? Wouldn't we not have made progress visible if the connection attempt was lower than `SHOW_POPUP_ON_ATTEMPT`?

Reply:
@nhooyr Did you mean "Wouldn't we not have made progress visible if the connection attempt was **greater** than `SHOW_POPUP_ON_ATTEMPT`?" Then yes, these calls are redundant. The code could be simplified to 
```typescript
		switch (e.type) {
			case PersistentConnectionEventType.ConnectionLost:
				hideProgress();
				break;
```
Or did you mean something else?

Reply:
Yes sorry, I didn't mean to add the extra `not` in there.

Yup in that case this looks good to me, will merge. I'll do a follow up commit later to simplify what I can.

Thanks again!

Reply:
Thanks a lot for the feedback and accepting the PR!